### PR TITLE
feat: Docker image

### PR DESCRIPTION
**Pull request by tyrossel**
- [ ] TODO: change the docker registry name in the GitHub deploy workflow

### CODE REVIEW

File: docker/nginx.conf.template [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  gzip_types application/javascript text/css application/json;
+
+  location / {
+    try_files $uri $uri/ =404;
+  }
+
+  location = / {
+    index $APP_PATH/index.html;
+  }
```
Comment (gberaudo) [inline docker/nginx.conf.template:21]: Could you add some comment about the difference between "/" and "= /"?

File: docker/nginx.conf.template [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  gzip on;
+
+  gzip_types application/javascript text/css application/json;
+
+  location / {
+    try_files $uri $uri/ =404;
+  }
+
+  location = / {
+    index $APP_PATH/index.html;
```
Comment (gberaudo) [inline docker/nginx.conf.template:20]: Here we should also change to expiration (https://nginx.org/en/docs/http/ngx_http_headers_module.html)
```suggestion
    index $APP_PATH/index.html;
    expires -1:
```

File: docker/14-ngv.envsh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+# This file will be sourced in the container before running the env substitution in the container.
+
+APP_NAME="${APP_NAME:-all}"
+
+APP_PATH="${APP_PATH:-/}"
+case "$APP_NAME" in
+"") ;;
+"/") ;;
+"all") ;;
+*) export APP_PATH="src/apps/${APP_NAME}" ;;
```
Comment (gberaudo) [inline docker/14-ngv.envsh:11]: It looks strange that the APP_PATH can finish, or not with a '/'.
Shouldn't it be empty by default?

File: Dockerfile [hunk 4]
```diff
@@ -1,11 +1,28 @@
...
+
+COPY --from=builder /app/dist /usr/share/nginx/html
+
+# The nginx.conf.template file is used to configure the nginx server.
+# In the entrypoint, the environment variables are automatically replaced.
+# See docs for more information: https://hub.docker.com/_/nginx
+COPY docker/*.conf.template /etc/nginx/templates/
+
+# Hooks for the entrypoint
+# env files are executed, envsh files are sources
```
Comment (gberaudo) [inline Dockerfile:27]:
```suggestion
# env files are executed, envsh files are sourced
```

**Review by gberaudo — APPROVED** [review]

File: docker/14-ngv.envsh (hunk 3 above)
Comment (tyrossel) [reply docker/14-ngv.envsh:10]: You're right, it's either empty or `src/apps/${APP_NAME}`

**Pull request merged**